tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -386,9 +386,13 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
                     var activity = Activity.ActivityListener.GetCurrentActivity();
                     if (activity is Activity.DuckTypes.IW3CActivity w3CActivity)
                     {
-                        // If there's an existing activity we use the same traceId (converted).
-                        rawTraceId = w3CActivity.TraceId;
-                        traceId = Convert.ToUInt64(w3CActivity.TraceId.Substring(16), 16);
+                        // If the Activity has an ActivityIdFormat.Hierarchical instead of W3C it's TraceId & SpanId will be null
+                        if (w3CActivity.TraceId is not null)
+                        {
+                            // If there's an existing activity we use the same traceId (converted).
+                            rawTraceId = w3CActivity.TraceId;
+                            traceId = Convert.ToUInt64(w3CActivity.TraceId.Substring(16), 16);
+                        }
```

Comment:
```suggestion
                        // If the Activity has an ActivityIdFormat.Hierarchical instead of W3C it's TraceId & SpanId will be null
                        if (w3CActivity.TraceId is { } w3cTraceId)
                        {
                            // If there's an existing activity we use the same traceId (converted).
                            rawTraceId = w3cTraceId;
                            traceId = Convert.ToUInt64(w3cTraceId.Substring(16), 16);
                        }
```

Reply:
👍